web/src/components/Header.js

```diff
@@ -6,37 +8,102 @@ import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { push } from 'connected-react-router';
 
-
+import { getIsAdmin } from '../reducers/user';
 import { t } from '../i18n';
 
+import Icon, { faChevronDown, faChevronLeft, faEdit, faSignOutAlt } from './Icons';
+
 class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ariaExpanded: props.ariaExpanded,
+    };
+    this.handleOutsideClick = this.handleOutsideClick.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleOutsideClick);
+  }
+
+  handleOutsideClick = e => {
+    if (this.node.contains(e.target)) {
+      return;
+    }
+    this.setState({ ariaExpanded: false })
+  };
+
   handleLogout = e => {
     e.preventDefault();
     const { pushRoute } = this.props;
     pushRoute('/logout');
   };
 
+  toggleDropdown = () => {
+    this.setState(prev => ({ ariaExpanded: !prev.ariaExpanded }));
+  };
+
   render() {
-    const { authenticated } = this.props;
+    const { authenticated, currentUser, isAdmin, showSiteTitle } = this.props;
+    const { ariaExpanded } = this.state;
     return (
-      <header className="header__slim">
+      <header ref={node => this.node = node}>
```

Comment:
This is some black magic right here. But it's what makes the `outsideClickHandler` work, so I'll take it.